File: Python/HBD_Debian.py
import math

# --- 1. Set Constants ---
WIDTH = 40
HEIGHT = 22
GOLDEN_RATIO = (1 + math.sqrt(5)) / 2
SPIRAL_CONST_B = math.log(GOLDEN_RATIO) / (math.pi / 2)
SPIRAL_CONST_A = 1.2

# --- 2. Define a Canvas ---
canvas = [[' ' for _ in range(WIDTH)] for _ in range(HEIGHT)]

# --- 3. Loop through the Angle (theta) ---
theta = 0.0
while theta < 6 * math.pi:
    radius = SPIRAL_CONST_A * math.exp(SPIRAL_CONST_B * theta)
    x = radius * math.cos(theta)
    y = -radius * math.sin(theta)
    screen_x = int(x * 2 + WIDTH / 2)
    screen_y = int(y + HEIGHT / 2)
    if 0 <= screen_x < WIDTH and 0 <= screen_y < HEIGHT:
        canvas[screen_y][screen_x] = '$'
    theta += 0.02

canvas[int(HEIGHT / 2)][int(WIDTH / 2)] = '@'

# --- 7. Print the Canvas ---
print("\n--- Happy Birthday Debian!! ---")
for row in reversed(canvas):
    print("".join(row[::-1]))
print("\n         Debian              ")
print("------------------------------------------- \n")

File: Kotlin/HBD_Debian.kt
import kotlin.math.cos
import kotlin.math.exp
import kotlin.math.ln
import kotlin.math.PI
import kotlin.math.sin
import kotlin.math.sqrt

// --- 1. Set Constants ---
// 'const val' is used for compile-time constants.
const val WIDTH = 40
const val HEIGHT = 22
const val SPIRAL_CONST_A = 1.2

// These are calculated at runtime, so they are 'val' not 'const val'.
val GOLDEN_RATIO = (1 + sqrt(5.0)) / 2
val SPIRAL_CONST_B = ln(GOLDEN_RATIO) / (PI / 2)

fun main() {
    // --- 2. Define a Canvas ---
    val canvas = MutableList(HEIGHT) { MutableList(WIDTH) { ' ' } }

    // --- 3. Loop through the Angle (theta) ---
    var theta = 0.0
    while (theta < 6 * PI) {
        val radius = SPIRAL_CONST_A * exp(SPIRAL_CONST_B * theta)
        val x = radius * cos(theta)
        val y = -radius * sin(theta)

        // Convert mathematical coordinates to screen coordinates
        val screenX = (x * 2 + WIDTH / 2).toInt()
        val screenY = (y + HEIGHT / 2).toInt()

        // Draw the spiral character if it's within the canvas bounds
        if (screenX in 0 until WIDTH && screenY in 0 until HEIGHT) {
            canvas[screenY][screenX] = '$'
        }
        theta += 0.02
    }

    // Place the center character
    canvas[HEIGHT / 2][WIDTH / 2] = '@'

    // --- 4. Print the Canvas ---
    println("\n--- Happy Birthday Debian!! ---")
    // Iterate through the canvas rows in reverse to print correctly
    for (row in canvas.reversed()) {
        // Reverse each row and join to a string to match the original output
        println(row.joinToString("").reversed())
    }
    println("\n         Debian              ")
    println("-------------------------------------------\n")
}
